### ISSUE TITLE
Excludes for hooks

### ISSUE THREAD

**Issue by LandonSchropp** [thoughtbot/rcm#187]
It would be great if `EXCLUDES` applied to hooks as well as linked files. For example, it would be awesome if I could do this in my `rcrc` file:

```
EXCLUDES="$HOME/.thoughtbot_dotfiles:hooks/*"
DOTFILES_DIRS="$HOME/.dotfiles $HOME/.thoughtbot_dotfiles"
```

**Comment by mike-burns** [comment]
For your patch, you'll want to look at the `run_hooks` function in `share/rcm.sh.in`, especially the call to find(1).

That said, what's the use case for this?

**Comment by LandonSchropp** [comment]
@mike-burns Thanks for the quick reply. Here's what my actual `rcrc` file looks like:

```
EXCLUDES=(
  "$HOME/.thoughtbot_dotfiles:README.md"
  "$HOME/.thoughtbot_dotfiles:LICENSE"
  "$HOME/.thoughtbot_dotfiles:aliases"
  "$HOME/.thoughtbot_dotfiles:zsh*"
  "$HOME/.thoughtbot_dotfiles:*vim*"
  "$HOME/.thoughtbot_dotfiles:hooks/*"
  "readme.md"
  "osx.zsh"
  "rcrc"
  "xterm"
)

EXCLUDES=${EXCLUDES[@]}
DOTFILES_DIRS="$HOME/.dotfiles $HOME/.thoughtbot_dotfiles"
```

As you can see, I'm excluding the Thoughtbot Vim files. One the files are symlinked, I'd like to install the plugins from my `.config/nvim/plugins.vim` file. To accomplish that, I added a `post-up` hook to my dotfiles repo:

``` shell
#!/bin/bash

# Download Plug
if [ ! -e "$HOME"/.config/nvim/autoload/plug.vim ]; then
  curl -fLo "$HOME"/.config/nvim/autoload/plug.vim --create-dirs \
      https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
fi

# Install the plugins listed in the plugins.vim file
nvim -u "$HOME"/.config/nvim/plugins.vim +PlugInstall +PlugClean! +qa
```

Unfortunately, post-up hook runs *before* the Thoughtbot dotfiles [post-up hook](https://github.com/thoughtbot/dotfiles/blob/master/hooks/post-up#L9). Since the Thoughtbot post-up runs `PlugClean`, my local plugins get wiped out.

As a temporary workaround, I've added this to my `post-up` file:

```
rm -rf $HOME/.thoughtbot_dotfiles/hooks
```

That's obviously not a good solution, so I was hoping for a quick way to prevent the Thoughtbot hooks from running.

**Comment by mat-m** [comment]
@LandonSchropp Can we consider this a duplicate of #102 , fixed for real with #272  ?

**Comment by LandonSchropp** [comment]
@mat-m Honestly, it's been a bit since I've looked at this, so sounds good to me. 🙂
